modules/scripts/startup-script/files/setup-raid.yml

```diff
@@ -0,0 +1,88 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+
+- name: Configure local SSDs
+  become: true
+  hosts: localhost
+  vars:
+    DST_MNT: "/mnt/localssd"
+    bashrc: "{{ '/etc/bashrc' if ansible_facts['os_family'] == 'RedHat' else '/etc/bash.bashrc' }}"
+    array_dev: "/dev/md0"
+    mode: '0755'
+  tasks:
+  - name: Check if {{ array_dev }} exists
+    ansible.builtin.stat:
+      path: "{{ array_dev }}"
+    register: array_dev_exists
+
+  - name: Install mdadm
+    ansible.builtin.package:
+      name: mdadm
+      state: present
+
+  - name: Get number of devices
+    ansible.builtin.shell: nvme list | { grep nvme_ || test $? = 1; }  | { grep -v nvme_card-pd || test $? = 1; } | awk '{print $1}' | wc -l
+    register: nvme_device_count
+
+  - name: Get list of devices
+    ansible.builtin.shell: nvme list | { grep nvme_ || test $? = 1; }  | { grep -v nvme_card-pd || test $? = 1; } | awk '{print $1}' | paste -sd ' '
+    register: nvme_device_list
+
+  - name: Exit if less than 2 local ssd
+    ansible.builtin.meta: end_play
+    when: nvme_device_count.stdout < "2"
+
+  - name: Check if there is an existing foreign RAID device
+    ansible.builtin.shell: lsblk | grep -q 'md127'
+    register: raid_check
+    failed_when: false
+    changed_when: false
+
+  - name: Stop and remove foreign RAID device
+    ansible.builtin.shell: mdadm --manage /dev/md127 --stop --remove
+    when: raid_check.rc == 0
+
+  - name: Zero superblock on foreign RAID device
+    ansible.builtin.shell:
+      cmd: sudo mdadm --zero-superblock {{ nvme_device_list.stdout }}
+    when: raid_check.rc == 0
+
+  - name: Create RAID-0 array if none were found
```

Comment:
If you follow the advice above to remove `stat` and add `find`, you can use the code below to achieve a similar effect

```yaml
              args:
                creates: "{{ array_dev }}"
              when: local_ssd_devices.files | length >= 2
```

The `mdadm` command you need to simulate is

```
mdadm --create /dev/md/raid_name --name=raid_name --homehost=any --level=0 --raid-devices=2 /dev/disk/by-id/google-local-nvme-ssd-*
```

The presence of the asterisk in this command will mean that you need to change from `ansible.builtin.command` to `ansible.builtin.shell` (which supports shell things like the fileglob asterisk)

Reply:
Changed to use `creates`